src/git.rs

```diff
@@ -1,263 +1,250 @@
-                line_start: 7,
-                line_end: 7,
+                file_name: "bar.rs".into(),
+                line_start: 5,
+                line_end: 9,
+            },
+            Section {
+                file_name: "foo.rs".into(),
+                line_start: 3,
+                line_end: 4,
             },
             Section {
-                file_name: "src/git.rs".to_string(),
-                line_start: 120,
-                line_end: 180,
+                file_name: "foo.rs".into(),
+                line_start: 6,
+                line_end: 7,
             },
         ];
-        let parser = Parser::new();
-        // Run
-        let actual_sections = parser.sections(&diff);
-        // Assert
-        assert_eq!(expected_sections, actual_sections);
+        let actual = get_sections(repo.path(), "master").unwrap();
+        assert_eq!(expected, actual);
+        Ok(())
+    }
+
+    #[test]
+    fn mixed_extensions() -> Result<()> {
+        let repo = RepoFixture::new()?
+            .write("foo.rs", "test_files/git/mixed/foo.rs")?
+            .write("bar.txt", "test_files/git/mixed/bar.txt")?
+            .stage(&["foo.rs", "bar.txt"])?;
+
+        let expected = vec![Section {
+            file_name: "foo.rs".into(),
+            line_start: 1,
+            line_end: 7,
+        }];
+        let actual = get_sections(repo.path(), "master").unwrap();
+        assert_eq!(expected, actual);
+        Ok(())
     }
+
     #[test]
-    fn test_diff_several_extensions_files() {
-        use crate::git::{Parser, Section};
-        // Setup
-        let diff = std::fs::read_to_string("test_files/git/diff_several_extensions_files.patch").unwrap();
-        let expected_sections: Vec<Section> = vec![
+    fn other_branch() -> Result<()> {
+        let repo = RepoFixture::new()?
+            .branch("other")?
+            .write("foo.rs", "test_files/git/modified/old/foo.rs")?
+            .stage(&["foo.rs"])?
+            .commit("other", &["foo.rs"])?
+            .write("foo.rs", "test_files/git/modified/new/foo.rs")?;
+
+        let expected = vec![
             Section {
-                file_name: "src/git.rs".to_string(),
-                line_start: 91,
-                line_end: 97,
+                file_name: "foo.rs".into(),
+                line_start: 3,
+                line_end: 4,
+            },
+            Section {
+                file_name: "foo.rs".into(),
+                line_start: 6,
+                line_end: 7,
             },
         ];
-        let parser = Parser::new();
-        // Run
-        let actual_sections = parser.sections(&diff);
-        // Assert
-        assert_eq!(expected_sections, actual_sections);
+        let actual = get_sections(repo.path(), "other").unwrap();
+        assert_eq!(expected, actual);
+        Ok(())
+    }
+
+    struct RepoFixture {
+        dir: TempDir,
+        repo: Repository,
+    }
+
+    impl RepoFixture {
```

Comment:
That's a great idea ! could we maybe add an impl Drop for RepoFixture to clean our mess once the test is done ?

Reply:
It seems that is already what happens when you drop a `TempDir` according to the [docs](https://docs.rs/tempfile/3.1.0/tempfile/struct.TempDir.html). I've checked and the directories seem to be removed :)

Reply:
Perfect then let's merge this !
Thank you so much for such a solid pull request !